### ISSUE TITLE
fix: markup and improve accessibility on the Tree component

### ISSUE THREAD

**Issue by TahimiLeonBravo** [nexxtway/react-rainbow#1669]
### Specs
- [x] It should render a `ul` element instead of `li` when an element has one or more children (here the console warns about that one li cannot contains li as children, then we need to find a good solution).
- [x] It should have a `span` element instead `h1` on the label element
- [ ] It should follow the accessibility standards on the markup
- [ ] It should make clickable all items on the tree
- [ ] It should work with the keyboard
- [ ] It should allow the user to do any action when clicking on the tree items

#### Design Guidelines
https://zpl.io/bAoorRB

#### Accessibility Sources
- https://www.w3.org/TR/wai-aria-practices/examples/treeview/treeview-2/treeview-2b.html
-https://www.lightningdesignsystem.com/components/trees/#site-main-content
